File: js/complete/no003.js
#!/usr/bin/env node

/* What is the largest prime factor of the number 600851475143 */

var fns = require('../functions.js'),
    timer = require('../timer.js');

exports.main = function(verbose) {
    if (typeof verbose == 'undefined') {
        verbose = false;
    }
    return Math.max.apply(Math, fns.primeFactors(600851475143));
};

if (require.main === module) {
    timer.timer(3, exports.main);
}

File: python_code/complete/no003.py
#!/usr/bin/env python

# What is the largest prime factor of the number 600851475143

from python_code.decorators import euler_timer
from python_code.functions import prime_factors

def main(verbose=False):
    return max(prime_factors(600851475143))

if __name__ == "__main__":
    print euler_timer(3)(main)(verbose=True)
